qiita_core/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,348 @@
+        self.assertEqual(obs.certificate_file, "/tmp/server.cert")
+        self.assertEqual(obs.cookie_secret, "SECRET")
+        self.assertEqual(obs.key_file, "/tmp/server.key")
+
+        # Postgres section
+        self.assertEqual(obs.user, "postgres")
+        self.assertEqual(obs.admin_user, "postgres")
+        self.assertEqual(obs.password, "andanotherpwd")
+        self.assertEqual(obs.admin_password, "thishastobesecure")
+        self.assertEqual(obs.database, "qiita_test")
+        self.assertEqual(obs.host, "localhost")
+        self.assertEqual(obs.port, 5432)
+
+        # Redis section
+        self.assertEqual(obs.redis_host, "localhost")
+        self.assertEqual(obs.redis_password, "anotherpassword")
+        self.assertEqual(obs.redis_db, 13)
+        self.assertEqual(obs.redis_port, 6379)
+
+        # SMTP section
+        self.assertEqual(obs.smtp_host, "localhost")
+        self.assertEqual(obs.smtp_port, 25)
+        self.assertEqual(obs.smtp_user, "qiita")
+        self.assertEqual(obs.smtp_password, "supersecurepassword")
+        self.assertFalse(obs.smtp_ssl)
+        self.assertEqual(obs.smtp_email, "example@domain.com")
+
+        # EBI section
+        self.assertEqual(obs.ebi_seq_xfer_user, "Webin-41528")
+        self.assertEqual(obs.ebi_seq_xfer_pass, "passwordforebi")
+        self.assertEqual(obs.ebi_seq_xfer_url, "webin.ebi.ac.uk")
+        self.assertEqual(
+            obs.ebi_dropbox_url,
+            "https://www-test.ebi.ac.uk/ena/submit/drop-box/submit/")
+        self.assertEqual(obs.ebi_center_name, "qiita-test")
+        self.assertEqual(obs.ebi_organization_prefix, "example_organization")
+
+        # IPython section
+        self.assertEqual(obs.ipython_contexts, ["qiita-general"])
+        self.assertEqual(obs.ipython_default, "qiita-general")
+
+        # VAMPS section
+        self.assertEqual(obs.vamps_user, "user")
+        self.assertEqual(obs.vamps_pass, "password")
+        self.assertEqual(obs.vamps_url,
+                         "https://vamps.mbl.edu/mobe_workshop/getfile.php")
+
+        # Portal section
+        self.assertEqual(obs.portal_fp, "/tmp/portal.cfg")
+        self.assertEqual(obs.portal, "QIITA")
+        self.assertEqual(obs.portal_dir, "/portal")
+
+    def test_init_error(self):
+        with open(self.conf_fp, 'w') as f:
+            f.write("\n")
+
+        with self.assertRaises(MissingConfigSection):
+            ConfigurationManager()
+
+    def test_get_main(self):
+        obs = ConfigurationManager()
+
+        conf_setter = partial(self.conf.set, 'main')
+        conf_setter('COOKIE_SECRET', '')
+        conf_setter('BASE_DATA_DIR', '')
+        conf_setter('PLUGIN_DIR', '')
+        conf_setter('CERTIFICATE_FILE', '')
+        conf_setter('KEY_FILE', '')
+
+        # Warning raised if No files will be allowed to be uploaded
+        # Warning raised if no cookie_secret
+        with warnings.catch_warnings(record=True) as warns:
```

Comment:
Does assertWarns allow you to check the contents of the warning? We have been using `npt.assert_warns` in other places of the code, but here I decided to use the catch warnings approach so I can test that the contents of the warnings is what I expect.